ledger/src/blockstore.rs

```diff
@@ -1484,28 +1476,39 @@ impl Blockstore {
         result
     }
 
-    fn find_conflicting_coding_shred(
-        &self,
+    fn find_conflicting_coding_shred<'a>(
+        &'a self,
         shred: &Shred,
         slot: Slot,
         erasure_meta: &ErasureMeta,
-        just_received_shreds: &HashMap<ShredId, Shred>,
-    ) -> Option<Vec<u8>> {
+        just_received_shreds: &'a HashMap<ShredId, Shred>,
+    ) -> Option<Cow<Vec<u8>>> {
         // Search for the shred which set the initial erasure config, either inserted,
         // or in the current batch in just_received_shreds.
+        let index = erasure_meta.first_received_coding_shred_index()?;
+        let shred_id = ShredId::new(slot, index, ShredType::Code);
+        let maybe_shred = self.get_shred_from_just_inserted_or_db(just_received_shreds, shred_id);
+
+        if index != 0 || maybe_shred.is_some() {
+            return maybe_shred;
+        }
```

Comment:
In this scenario:
* shred 0 had a conflict with shred 5, we would've stored the duplicate proof when inserting 0
* shred 0 and shred 5 are consistent, when inserting the new shred with v1.18.12 both shred 0 or shred 5 are fine for the duplicate proof since they produce the same erasure meta

Reply:
Ah ok, so we basically have the transitive property for "conflicting" here.
```
1) shred0 and shred5 are consistent
2) shred5 and shred7 are inconsistent

1) and 2) imply that shred0 and shred7 are inconsistent
```

Reply:
yep exactly, this specific check compares to see if the ErasureMeta generated by the shred is consistent. Well formed coding shreds from the same FEC set should produce the same ErasureMeta.